lua/toggleterm.lua

```diff
@@ -184,6 +184,80 @@ function M.exec(cmd, num, size, dir, direction, go_back, open)
   term:send(cmd, go_back)
 end
 
+--- @param selection_type string
+--- @param trim_spaces boolean
+--- @param terminal_id number
+function M.send_lines_to_terminal(selection_type, trim_spaces, terminal_id)
+  -- trim_spaces defines if we should trim the spaces from lines which are sent to the terminal
+  if trim_spaces == nil then trim_spaces = true end
+
+  -- If no terminal id provided fall back to the default
+  if terminal_id == nil then terminal_id = 1 end
+  terminal_id = tonumber(terminal_id)
+
+  vim.validate({
+    selection_type = { selection_type, "string", true },
+    trim_spaces = { trim_spaces, "boolean", true },
+    terminal_id = { terminal_id, "number", true }
+  })
+
+  -- Window number from where we are calling the function (needed so we can get back to it automatically)
+  local current_window = vim.api.nvim_get_current_win()
+  -- Line texts - these will be sent over to the terminal one by one
+  local lines = {}
+  -- Beginning of the selection: line number, column number
+  local b_line, b_col
+
+  local function _line_selection(mode)
+    local start_char, end_char
+    if mode == "visual" then
+      start_char = "'<"
+      end_char = "'>"
+    elseif mode == "motion" then
+      start_char = "'["
+      end_char = "']"
+    end
+
+    -- Get the start and the end of the selection
+    local start_line, start_col = unpack(vim.fn.getpos(start_char), 2, 3)
+    local end_line, end_col = unpack(vim.fn.getpos(end_char), 2, 3)
+    local selected_lines = vim.api.nvim_buf_get_lines(0, start_line - 1, end_line, 0)
+    return {start_pos={start_line, start_col}, end_pos={end_line, end_col}, selected_lines=selected_lines}
+  end
+
+  if selection_type == "visual_lines" or selection_type == "visual_selection" then
+    local res = _line_selection("visual")
```

Comment:
nit: don't think the `_` prefix is necessary here since it's already local to this function

Reply:
Done